docs/ABI.rst

```diff
@@ -877,8 +877,8 @@ Entities
   curry-thunk ::= 'Tc'
 
   // The leading type is the function type
-  entity-spec ::= type 'fC'                  // allocating constructor
-  entity-spec ::= type 'fc'                  // non-allocating constructor
+  entity-spec ::= file-discriminator? type 'fC'      // allocating constructor
+  entity-spec ::= file-discriminator? type 'fc'      // non-allocating constructor
```

Comment:
It's better to add the optional discriminator after the type:

    entity-spec ::= type file-discriminator? 'fc'

Having optional terms at the beginning of a rule can be problematic because when demangling it's not clear if they belong to that rule or to then "enclosing" rule.
For the file-discriminator it's actually not a problem because it's not used elsewhere. But to be consistent (and future prove) I'd prefer to change it.

Alternatively you could also just add a new character to 'f' instead of file-discriminator

    entity-spec ::= type identifier 'fB' // allocating file-discriminated constructor
    entity-spec ::= type identifier 'fb' // non-allocating file-discriminated constructor